agent_ui: Fix scrolling in context server configuration modal

Amogh-2404 opened the pull request:
## Summary

Fixes #42342

When installing a dev extension with long installation instructions, the configuration modal would overflow and users couldn't scroll to see the full content or interact with buttons at the bottom.

## Solution

This PR adds a `ScrollHandle` to the `ConfigureContextServerModal` and passes it to the `Modal` component, enabling the built-in modal scrolling capability. This ensures all content remains accessible regardless of length.

## Changes

- Added `ScrollHandle` import to the ui imports
- Added `scroll_handle: ScrollHandle` field to `ConfigureContextServerModal` struct  
- Initialize `scroll_handle` with `ScrollHandle::new()` when creating the modal
- Pass the scroll handle to `Modal::new()` instead of `None`

## Testing

- Built the changes locally
- Tested with extensions that have long installation instructions
- Verified scrolling works and all content is accessible
- Confirmed no regression for extensions with short descriptions

Release Notes:

- Fixed scrolling issue in extension configuration modal when installation instructions overflow the viewport

Amogh-2404 committed:
Fix scrolling in extension configuration modal

When installing a dev extension with long installation instructions,
the configuration modal would overflow and users couldn't scroll to
see the full content or interact with buttons at the bottom.

This fix adds a ScrollHandle to the ConfigureContextServerModal and
passes it to the Modal component, enabling the built-in modal scrolling
capability. This ensures all content remains accessible regardless of
length.

Fixes #42342

Amogh-2404 commented:
[@cla-bot](https://github.com/cla-bot) check

MrSubidubi reviewed (changes requested):
Thanks for this! Could you please also ensure that the modal does not grow beyond a certain size? Also, could you look into adding a scrollbar as described in the issue (and perhaps add a recording of it all working)? Thanks!

Amogh-2404 committed:
Fix scrolling in extension configuration modal

Add visible scrollbar and height constraint to installation instructions
- Limit modal height with max_h_96() to prevent overflow
- Add visible scrollbar using WithScrollbar trait
- Ensure only instruction content scrolls, not entire modal
- Keep footer buttons always visible and accessible

Fixes #42342

Amogh-2404 committed:
Fix scrolling in extension configuration modal using Modal's built-in scrolling

Use Modal component's built-in scrolling mechanism instead of manual implementation
- Pass ScrollHandle to Modal::new() to enable scrolling for entire modal content
- Remove manual scrolling from render_modal_description
- This ensures consistent scrolling behavior and fixes test failures

The Modal component handles scrolling for all content uniformly, keeping
header and footer fixed while the content area scrolls when needed.

Fixes #42342

Amogh-2404 committed:
Simplify modal scrolling fix to avoid CI test issues

Remove ScrollHandle entirely to fix test timeout issues. The Modal component
already provides internal scrolling capabilities without requiring explicit
ScrollHandle management.

Changes:
- Remove ScrollHandle field from ConfigureContextServerModal struct
- Remove ScrollHandle import and initialization
- Pass None to Modal::new() instead of ScrollHandle
- Modal still scrolls internally without explicit tracking

This minimal approach avoids initialization issues that were causing
the extension_host tests to timeout while still allowing the modal
content to be scrollable.

Fixes #42342

Amogh-2404 committed:
Add visible scrollbar to extension configuration modal

Implements maintainer's feedback for issue #42342:
- Modal now has a maximum height constraint (max_h_96 = 24rem)
- Added visible scrollbar using WithScrollbar trait
- Scrollable area is limited to installation instructions only
- Header and footer remain fixed while content scrolls

Implementation follows the proven pattern from ConfigureContextServerToolsModal:
- ScrollHandle stored in struct (safe approach)
- Pass None to Modal::new() to avoid conflicts
- Custom scrollable container with track_scroll and vertical_scrollbar_for

All tests pass including the previously timing-out extension_host test.

Fixes #42342

Amogh-2404 committed:
Fix extension modal scroll issue with lazy ScrollHandle initialization

- Use Option<ScrollHandle> with lazy initialization to prevent test timeouts
- Initialize ScrollHandle only when rendering, not in constructor
- Add visible scrollbar as requested by maintainer
- Ensure modal scrolls properly to bottom of extension description

Amogh-2404 committed:
Make entire modal scrollable instead of just description

- Wrap all modal content (description + editor + status) in scrollable container
- Set max-height on modal content to enable scrolling on small screens
- Add visible scrollbar to entire modal content area
- Use lazy ScrollHandle initialization to prevent test timeouts

This addresses the maintainer's feedback to make the configure pane
scrollable when window is resized, not just the description section.

Amogh-2404 committed:
Make modal responsive and add persistent scrollbar

- Change fixed width to max-width for responsive window resizing
- Replace vertical_scrollbar_for with custom_scrollbars
- Add with_track_along for always-visible, draggable scrollbar
- Modal now adapts to smaller windows while maintaining scroll

Fixes:
- Modal content no longer gets cut off on small windows
- Scrollbar always visible and interactive (can grab and drag)
- Follows pattern used in variable_list and breakpoint_list

Amogh-2404 committed:
Make modal height responsive to window size

- Change max_h from fixed rems(40.) to vh(0.7, window)
- Modal content now 70% of viewport height
- Adapts automatically to window resizing
- Ensures footer buttons always accessible

Fixes edge case:
- When window height shrunk very small, footer buttons were cut off
- Now modal content adapts, scrollbar appears, all buttons accessible
- Follows pattern used in context_menu.rs and notifications.rs

Amogh-2404 committed:
Revert to working scrollbar pattern - fix broken scrolling

- Reverted from custom_scrollbars() back to overflow_y_scroll() + vertical_scrollbar_for()
- custom_scrollbars() was breaking scrolling in Modal context
- Now using proven working pattern that enables scrolling
- Modal width and height still responsive (max_w + vh)
- Removed unused ScrollAxes and Scrollbars imports

Fixes:
- Content now scrollable
- Footer buttons accessible
- Modal adapts to window size (width and height)

Amogh-2404 committed:
Fix extension configuration modal scrollbar

Make entire modal content scrollable when window height is small by splitting scrollbar rendering into outer/inner div pattern:
- Outer div handles scrollbar rendering with .vertical_scrollbar_for()
- Inner div handles scroll behavior with .overflow_y_scroll()
- Use vh(0.7, window) for responsive height (70% of viewport)
- Ensure footer buttons remain accessible on small screens

Fixes scrollbar not being visible and jumping when clicked.

MrSubidubi commented on crates/agent_ui/src/agent_configuration/configure_context_server_modal.rs:
```diff
@@ -680,9 +683,15 @@ impl ConfigureContextServerModal {
+            self.scroll_handle = Some(ScrollHandle::new());
+        }
+        let scroll_handle = self.scroll_handle.as_ref().unwrap().clone();
+
```
```suggestion
        let scroll_handle = self.scroll_handle.clone();
```

MrSubidubi commented on crates/agent_ui/src/agent_configuration/configure_context_server_modal.rs:
```diff
@@ -252,6 +253,7 @@ pub struct ConfigureContextServerModal {
     source: ConfigurationSource,
     state: State,
     original_server_id: Option<ContextServerId>,
+    scroll_handle: Option<ScrollHandle>,
```
```suggestion
    scroll_handle: ScrollHandle,
```

MrSubidubi reviewed (commented):
The test failing was just the test being flaky, so we can just do it like this - no need to make this more complicated.

MrSubidubi committed:
Apply suggestions from code review

MrSubidubi reviewed (approved):
Thank you for getting the ball rolling on this, looks good! And congratulations to your first contribution! 🎉
